### PR TITLE
Refactor `db` module to use strongly typed errors.

### PR DESCRIPTION
**Pull request by ryzhyk**
Replace all uses of `anyhow::Error` in `pipeline_manager::db` with strongly typed errors.

### CODE REVIEW

**Comment by mihaibudiu** [comment]
Is DBError a good name? Suggests something is wrong with a database.

File: crates/pipeline_manager/src/db/mod.rs [hunk 1]
```diff
@@ -122,6 +122,31 @@ impl Display for Version {
 #[derive(Debug, Serialize)]
 #[serde(untagged)]
 pub(crate) enum DBError {
+    #[serde(serialize_with = "serialize_pg_error")]
```
Comment (lalithsuresh) [inline crates/pipeline_manager/src/db/mod.rs:125]: Let's move everything related to errors to a separate file and module. db/mod.rs is getting crowded.

File: crates/pipeline_manager/src/runner.rs [hunk 2]
```diff
@@ -305,7 +305,7 @@ impl LocalRunner {
                     .await
                 {
                     let _ = pipeline_process.kill().await;
-                    return Err(e);
+                    bail!(e);
```
Comment (lalithsuresh) [inline crates/pipeline_manager/src/runner.rs:308]: do we still need bail!() now that we're getting rid of anyhow?

**Review by lalithsuresh — APPROVED** [review]

File: crates/pipeline_manager/src/db/mod.rs (hunk 1 above)
Comment (gz) [reply crates/pipeline_manager/src/db/mod.rs:125]: lets do it after we merged all the outstanding PRs though it will be a lot of merge conflict

**Review by gz — APPROVED** [review]

**Comment by ryzhyk** [comment]
I'll merge this and rebase the versioning PR on top of it straight away.

**Pull request merged**